diff --git a/rmf_traffic/include/rmf_traffic/schedule/Database.hpp b/rmf_traffic/include/rmf_traffic/schedule/Database.hpp
index 6318012..2d27b4b 100644
--- a/rmf_traffic/include/rmf_traffic/schedule/Database.hpp
+++ b/rmf_traffic/include/rmf_traffic/schedule/Database.hpp
@@ -113,7 +113,7 @@ public:
 
   // Documentation inherited from Viewer
   std::shared_ptr<const ParticipantDescription> get_participant(
-    std::size_t participant_id) const final;
+    ParticipantId participant_id) const final;
 
   // Documentation inherited from Viewer
   Version latest_version() const;
@@ -125,11 +125,11 @@ public:
 
   // Documentation inherited from ItineraryViewer
   std::optional<ItineraryView> get_itinerary(
-    std::size_t participant_id) const final;
+    ParticipantId participant_id) const final;
 
   // Documentation inherited from ItineraryViewer
   std::optional<PlanId> get_current_plan_id(
-    std::size_t participant_id) const final;
+    ParticipantId participant_id) const final;
 
   // Documentation inherited from ItineraryViewer
   const std::vector<CheckpointId>* get_current_progress(
diff --git a/rmf_traffic/include/rmf_traffic/schedule/Mirror.hpp b/rmf_traffic/include/rmf_traffic/schedule/Mirror.hpp
index 14c3c9e..f6a6108 100644
--- a/rmf_traffic/include/rmf_traffic/schedule/Mirror.hpp
+++ b/rmf_traffic/include/rmf_traffic/schedule/Mirror.hpp
@@ -51,11 +51,11 @@ public:
 
   // Documentation inherited from Viewer
   std::shared_ptr<const ParticipantDescription> get_participant(
-    std::size_t participant_id) const final;
+    ParticipantId participant_id) const final;
 
   // Documentation inherited from Viewer
   std::optional<ItineraryView> get_itinerary(
-    std::size_t participant_id) const final;
+    ParticipantId participant_id) const final;
 
   // Documentation inherited from Viewer
   std::optional<Version> latest_version() const;
diff --git a/rmf_traffic/src/rmf_traffic/schedule/Database.cpp b/rmf_traffic/src/rmf_traffic/schedule/Database.cpp
index 3f6dd48..2377692 100644
--- a/rmf_traffic/src/rmf_traffic/schedule/Database.cpp
+++ b/rmf_traffic/src/rmf_traffic/schedule/Database.cpp
@@ -1303,7 +1303,7 @@ const std::unordered_set<ParticipantId>& Database::participant_ids() const
 
 //==============================================================================
 std::shared_ptr<const ParticipantDescription> Database::get_participant(
-  std::size_t participant_id) const
+  ParticipantId participant_id) const
 {
   const auto state_it = _pimpl->descriptions.find(participant_id);
   if (state_it == _pimpl->descriptions.end())
@@ -1320,7 +1320,7 @@ Version Database::latest_version() const
 
 //==============================================================================
 std::optional<ItineraryView> Database::get_itinerary(
-  const std::size_t participant_id) const
+  const ParticipantId participant_id) const
 {
   const auto state_it = _pimpl->states.find(participant_id);
   if (state_it == _pimpl->states.end())
@@ -1345,7 +1345,7 @@ std::optional<ItineraryView> Database::get_itinerary(
 
 //==============================================================================
 std::optional<PlanId> Database::get_current_plan_id(
-  const std::size_t participant_id) const
+  const ParticipantId participant_id) const
 {
   const auto state_it = _pimpl->states.find(participant_id);
   if (state_it == _pimpl->states.end())
diff --git a/rmf_traffic/src/rmf_traffic/schedule/Mirror.cpp b/rmf_traffic/src/rmf_traffic/schedule/Mirror.cpp
index 4c1d710..e84006a 100644
--- a/rmf_traffic/src/rmf_traffic/schedule/Mirror.cpp
+++ b/rmf_traffic/src/rmf_traffic/schedule/Mirror.cpp
@@ -255,7 +255,7 @@ const std::unordered_set<ParticipantId>& Mirror::participant_ids() const
 
 //==============================================================================
 std::shared_ptr<const ParticipantDescription> Mirror::get_participant(
-  std::size_t participant_id) const
+  ParticipantId participant_id) const
 {
   const auto p = _pimpl->descriptions.find(participant_id);
   if (p == _pimpl->descriptions.end())
@@ -272,7 +272,7 @@ std::optional<Version> Mirror::latest_version() const
 
 //==============================================================================
 std::optional<ItineraryView> Mirror::get_itinerary(
-  const std::size_t participant_id) const
+  const ParticipantId participant_id) const
 {
   const auto p = _pimpl->states.find(participant_id);
   if (p == _pimpl->states.end())
@@ -297,7 +297,7 @@ std::optional<ItineraryView> Mirror::get_itinerary(
 
 //==============================================================================
 std::optional<PlanId> Mirror::get_current_plan_id(
-  const std::size_t participant_id) const
+  const ParticipantId participant_id) const
 {
   const auto p = _pimpl->states.find(participant_id);
   if (p == _pimpl->states.end())
