diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/broadphase/default_broadphase_callbacks.h b/rmf_traffic/thirdparty/fcl/include/fcl/broadphase/default_broadphase_callbacks.h
index 1c9b4fe..f6655fe 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/broadphase/default_broadphase_callbacks.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/broadphase/default_broadphase_callbacks.h
@@ -38,6 +38,7 @@
 #define FCL_BROADPHASE_DEFAULTBROADPHASECALLBACKS_H
 
 #include "fcl/narrowphase/collision.h"
+#include <cassert>
 #include "fcl/narrowphase/collision_request.h"
 #include "fcl/narrowphase/collision_result.h"
 #include "fcl/narrowphase/continuous_collision.h"
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/geometry/octree/octree-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/geometry/octree/octree-inl.h
index f50fe81..9b7f15a 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/geometry/octree/octree-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/geometry/octree/octree-inl.h
@@ -39,6 +39,7 @@
 #define FCL_OCTREE_INL_H
 
 #include "fcl/geometry/octree/octree.h"
+#include <cassert>
 
 #include "fcl/config.h"
 
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/geometry/shape/convex-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/geometry/shape/convex-inl.h
index 10adc69..0693975 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/geometry/shape/convex-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/geometry/shape/convex-inl.h
@@ -40,6 +40,7 @@
 #define FCL_SHAPE_CONVEX_INL_H
 
 #include <map>
+#include <cassert>
 #include <set>
 #include <utility>
 
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/math/motion/taylor_model/taylor_model-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/math/motion/taylor_model/taylor_model-inl.h
index 861f72d..bdaa3f4 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/math/motion/taylor_model/taylor_model-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/math/motion/taylor_model/taylor_model-inl.h
@@ -42,6 +42,7 @@
 #define FCL_CCD_TAYLOR_MODEL_INL_H
 
 #include "fcl/math/motion/taylor_model/taylor_model.h"
+#include <cassert>
 
 namespace fcl
 {
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/math/rng-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/math/rng-inl.h
index 1ba9da7..df18cd1 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/math/rng-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/math/rng-inl.h
@@ -39,6 +39,7 @@
 #define FCL_MATH_RNG_INL_H
 
 #include "fcl/math/rng.h"
+#include <cassert>
 
 namespace fcl
 {
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/convexity_based_algorithm/gjk_libccd-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/convexity_based_algorithm/gjk_libccd-inl.h
index 60fd0ad..51f63fe 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/convexity_based_algorithm/gjk_libccd-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/convexity_based_algorithm/gjk_libccd-inl.h
@@ -39,6 +39,7 @@
 #define FCL_NARROWPHASE_DETAIL_GJKLIBCCD_INL_H
 
 #include "fcl/narrowphase/detail/convexity_based_algorithm/gjk_libccd.h"
+#include <cassert>
 #include "fcl/narrowphase/detail/failed_at_this_configuration.h"
 
 #include <array>
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/capsule_capsule-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/capsule_capsule-inl.h
index b316c95..53e8532 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/capsule_capsule-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/capsule_capsule-inl.h
@@ -39,6 +39,7 @@
 #define FCL_NARROWPHASE_DETAIL_CAPSULECAPSULE_INL_H
 
 #include "fcl/narrowphase/detail/primitive_shape_algorithm/capsule_capsule.h"
+#include <cassert>
 
 namespace fcl
 {
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/plane-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/plane-inl.h
index 1373dce..1628214 100755
--- a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/plane-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/plane-inl.h
@@ -39,6 +39,7 @@
 #define FCL_NARROWPHASE_DETAIL_PLANE_INL_H
 
 #include "fcl/narrowphase/detail/primitive_shape_algorithm/plane.h"
+#include <cassert>
 
 namespace fcl
 {
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/sphere_box-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/sphere_box-inl.h
index 77e0a66..976ac2b 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/sphere_box-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/sphere_box-inl.h
@@ -38,6 +38,7 @@
 #define FCL_NARROWPHASE_DETAIL_SPHEREBOX_INL_H
 
 #include "fcl/narrowphase/detail/primitive_shape_algorithm/sphere_box.h"
+#include <cassert>
 
 namespace fcl {
 namespace detail {
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/sphere_cylinder-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/sphere_cylinder-inl.h
index abda262..8721550 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/sphere_cylinder-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/primitive_shape_algorithm/sphere_cylinder-inl.h
@@ -38,6 +38,7 @@
 #define FCL_NARROWPHASE_DETAIL_SPHERECYLINDER_INL_H
 
 #include "fcl/narrowphase/detail/primitive_shape_algorithm/sphere_cylinder.h"
+#include <cassert>
 
 namespace fcl {
 namespace detail {
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/traversal/distance/mesh_conservative_advancement_traversal_node-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/traversal/distance/mesh_conservative_advancement_traversal_node-inl.h
index 44ddde5..f86a123 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/traversal/distance/mesh_conservative_advancement_traversal_node-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/detail/traversal/distance/mesh_conservative_advancement_traversal_node-inl.h
@@ -39,6 +39,7 @@
 #define FCL_TRAVERSAL_MESHCONSERVATIVEADVANCEMENTTRAVERSALNODE_INL_H
 
 #include "fcl/narrowphase/detail/traversal/distance/mesh_conservative_advancement_traversal_node.h"
+#include <cassert>
 
 #include "fcl/math/bv/RSS.h"
 #include "fcl/math/motion/triangle_motion_bound_visitor.h"
diff --git a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/distance-inl.h b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/distance-inl.h
index 115b710..fb23949 100644
--- a/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/distance-inl.h
+++ b/rmf_traffic/thirdparty/fcl/include/fcl/narrowphase/distance-inl.h
@@ -39,6 +39,7 @@
 #define FCL_DISTANCE_INL_H
 
 #include "fcl/narrowphase/distance.h"
+#include <cassert>
 
 #include "fcl/narrowphase/collision.h"
 
